src/snowflake/snowpark/files.py

```diff
@@ -62,6 +112,22 @@ def __init__(
         self.encoding = None
         self.errors = None
 
+        _DEFAULT_READ_BUFFER_SIZE = 32 * 1024
+        if mode in READ_MODES:
+            # Buffered Reader used to support BufferedIOBase methods such as read1 and readinto1
+            self._file_stream = BufferedReader(
+                open(self._file_location, self._mode), _DEFAULT_READ_BUFFER_SIZE
+            )
+        elif mode in WRITE_MODES:
+            # need to still open a file stream
+            self._file_stream = open(self._file_location, self._mode)
+        self._pos = 0
+        self._is_local_file = (
+            True
+            if self._file_location.startswith((RELATIVE_PATH_PREFIX, "C:"))
```

Comment:
is this `_is_local_file` for local testing only and "C:" for windows OS?
if it's windows, users will only read files from disc C?